### PR TITLE
[Issue 33]: Implement admin layout

### PR DESCRIPTION
**Pull request by khoadev2512**
## Issue
- Connect #ISSUE_NUMBER ([Related Issue)](https://github.com/HopeeSolutionsRuby/demo-app/issues/33)

## ToDo
- [x] Task No.1:  Implement header
- [x] Task No.2: Implement sidebar 

### Coding Style
- [x] Check rubocop

## Test Result
- screenshot:
image

# ** Note
link test: http://localhost:3000/administrator/dashboard

### CODE REVIEW

**Review by hothanhdatsd — APPROVED** [review]

**Review by PhiH-XDev — APPROVED** [review]

**Review by minhnguyennn — APPROVED** [review]

File: Gemfile [hunk 1]
```diff
@@ -95,3 +95,6 @@ group :test do
   gem 'capybara'
   gem 'selenium-webdriver'
 end
+gem 'font-awesome-sass', '~> 6.5.1'
+
+gem 'devise', '~> 4.9'
```
Comment (KhoaNDAHopee) [inline Gemfile]: For what?

File: app/assets/stylesheets/administrator/header.scss
Comment (KhoaNDAHopee) [inline app/assets/stylesheets/administrator/header.scss:1]: Check spacing, endline

File: app/assets/stylesheets/fonts.scss
Comment (KhoaNDAHopee) [inline app/assets/stylesheets/fonts.scss:1]: Check spacing

File: app/controllers/administrator_controller.rb
Comment (KhoaNDAHopee) [inline app/controllers/administrator_controller.rb:1]: For what?

File: app/javascript/controllers/admin_controller.js
Comment (KhoaNDAHopee) [inline app/javascript/controllers/admin_controller.js:1]: Check spacing, endline

File: app/views/administrator/dashboard/_body.html.erb [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<div class="container-fluid">
+  <div class="row flex-nowrap">
+      <div class="collapse-show collapse-horizontal" id="collapseExample">
+        <div class="col-auto col-md-3 col-xl-2 px-sm-2 px-0 bg-light d-none d-sm-block sidebar" style="width: 250px;position:absolute; z-index:2 ;margin-left:-12px;">
+            <div class="d-flex flex-column align-items-start align-items-sm-start px-3 pt-2 text-white min-vh-100">
+                <ul class="nav nav-pills flex-column mb-sm-auto mb-0 align-items-center align-items-sm-start w-100" id="menu">
+                  <li class="nav-item w-100">
+                    <a href="#" class="nav-link px-0 align-middle w-100">
+                        <i class="fa-solid fa-user-doctor" style="margin-left: 3px;"></i> <span class="ms-2 d-sm-inline">Doctors</span></a>
```
Comment (KhoaNDAHopee) [inline app/views/administrator/dashboard/_body.html.erb]: Check spacing and closing tags
```
<a>
  abc
</a>
or
<a>abc</a>
```

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]

File: app/javascript/controllers/admin_controller.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    document.querySelector('.navbar-toggler').addEventListener('click', function () {
+      document.querySelector('.sidebar').classList.toggle('d-none');
+      document.querySelector('.sidebar').style.width = (document.querySelector('.sidebar').style.width === '250px') ? '100%' : '250px';
+      document.querySelector('.nav-footer').classList.toggle('d-none');
+      document.querySelector('.body-content').classList.toggle('d-none');
+  });
```
Comment (KhoaNDAHopee) [inline app/javascript/controllers/admin_controller.js]: Spacing

File: app/views/layouts/admin.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  </head>
+
+  <body>
+    <div class="admin-layout">
+      <div><%= render 'shared/administrator/header' %></div>
+        <div class="container-fluid">
+          <div class="row flex-nowrap">
+            <%= render 'shared/administrator/sidebar' %>
+            <div class="px-0 body-content" style="position: relative; width: 100%;">
+              <div style="min-height: 90vh; width: 100%">
```
Comment (KhoaNDAHopee) [inline app/views/layouts/admin.html.erb]: Don't use inline css

File: app/views/shared/administrator/_footer.html.erb [hunk 5]
```diff
@@ -0,0 +1 @@
+<div style="position: absolute; bottom: 0;border-top: 1px solid #dbdee4; padding-bottom:7px;padding-top:7px; text-align:center" class="w-100">Thank you for creating with Phoenix|2023 ©Themewagon </div>
```
Comment (KhoaNDAHopee) [inline app/views/shared/administrator/_footer.html.erb]: Don't use inline css

**Review by KhoaNDAHopee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**